src/main/java/uk/gov/nationalarchives/pdi/step/atomics/await/AwaitStepDialog.java

```diff
@@ -231,8 +234,30 @@ public void widgetSelected(final SelectionEvent selectionEvent) {
                 wInitialiseAtomicBooleanField.setVisible(ActionIfNoAtomic.Initialise == actionIfNoAtomic && atomicType == AtomicType.Boolean);
                 wInitialiseAtomicIntegerField.setVisible(ActionIfNoAtomic.Initialise == actionIfNoAtomic && atomicType == AtomicType.Integer);
 
-                wAtomicValueBooleanField.setVisible(atomicType == AtomicType.Boolean);
-                wAtomicValueIntegerField.setVisible(atomicType == AtomicType.Integer);
+                for (int rowIdx = 0; rowIdx < wAwaitTableView.getItemCount(); rowIdx++) {
+                    String atomicValue = wAwaitTableView.getItem(rowIdx, 1);
+                    atomicValue = strNullIfNull(atomicValue);
+
+                    if (!"null".equals(atomicValue)) {
+                        if (AtomicType.Boolean == atomicType) {
+                            atomicValue = unknownStrToBooleanStr(atomicValue);
+
+                        } else if (AtomicType.Integer == atomicType) {
+                            atomicValue = unknownStrToIntegerStr(atomicValue);
+                        }
+                    }
+
+                    wAwaitTableView.setText(atomicValue, 1, rowIdx);
+                }
+
+                // TODO(AR) can't figure out how to dynamically change the types of the table columns
+                /*
+                // use CCOMBO for true/false
+                ciAtomicValue.setFieldTypeColumn(atomicType == AtomicType.Boolean ? ColumnInfo.COLUMN_TYPE_CCOMBO : ColumnInfo.COLUMN_TYPE_TEXT);
+                wAwaitTableView.setColumnInfo(0, ciAtomicValue);
+                wAwaitTableView.redraw();
+                wAwaitTableView.update();
+                */
```

Comment:
Maybe there should be another ticket for this if this more work to do?

Reply:
Possibly. At the moment it isn't a serious issue, I am not even sure it is possible to do what I was trying to do. I am perhaps a little bit too cautious about keeping code around ;-)
Should I just remove this?